pkg/framework/machinesets.go

```diff
@@ -369,6 +459,125 @@ func WaitForMachineSetDelete(c runtimeclient.Client, machineSet *machinev1.Machi
+				if m.Status.ErrorReason != nil {
+					reason = string(*m.Status.ErrorReason)
+				}
+				message := "failureMessage not present in Machine.status"
+				if m.Status.ErrorMessage != nil {
+					message = string(*m.Status.ErrorMessage)
+				}
+				klog.Errorf("Failed machine: %s, Reason: %s, Message: %s", m.Name, reason, message)
+			}
+			return false, errMachineInMachineSetFailed
+		}
+
+		// Check if any machine did not get provisioned because of insufficient spot capacity.
+		for _, m := range machines {
+			insufficientCapacityResult, err := hasInsufficientCapacity(m, platform)
+			if err != nil {
+				return false, fmt.Errorf("error checking if machine %s has insufficient capacity: %v", m.Name, err)
+			}
+			if insufficientCapacityResult {
+				return false, ErrMachineNotProvisionedInsufficientCloudCapacity
+			}
+		}
+
+		running := FilterRunningMachines(machines)
+		// This could probably be smarter, but seems fine for now.
+		if len(running) != len(machines) {
+			klog.Infof("%q: not all Machines are running: %d of %d", name, len(running), len(machines))
+			return false, nil
+		}
+
+		for _, m := range running {
+			node, err := GetNodeForMachine(c, m)
+			if err != nil {
+				klog.Infof("Node for machine %s not found yet: %v", m.Name, err)
+				return false, nil
+			}
+
+			if !IsNodeReady(node) {
+				klog.Infof("%s: node is not ready", node.Name)
+				return false, nil
+			}
+		}
+
+		return true, nil
+	})
+	return err
+}
+
+// hasInsufficientCapacity return true if the machine cannot be provisioned due to insufficient spot capacity.
+func hasInsufficientCapacity(m *machinev1.Machine, platform configv1.PlatformType) (bool, error) {
+	switch platform {
+	case configv1.AWSPlatformType:
+		awsProviderStatus := machinev1.AWSMachineProviderStatus{}
+		if m.Status.ProviderStatus != nil {
+			err := json.Unmarshal(m.Status.ProviderStatus.Raw, &awsProviderStatus)
+			if err != nil {
+				return false, fmt.Errorf("error unmarshalling provider status: %v", err)
+			}
+			return hasInsufficientCapacityCondition(awsProviderStatus.Conditions, configv1.AWSPlatformType)
+		}
+	case configv1.AzurePlatformType:
+		azureProviderStatus := machinev1.AzureMachineProviderStatus{}
+		if m.Status.ProviderStatus != nil {
+			err := json.Unmarshal(m.Status.ProviderStatus.Raw, &azureProviderStatus)
+			if err != nil {
+				return false, fmt.Errorf("error unmarshalling provider status: %v", err)
+			}
+			return hasInsufficientCapacityCondition(azureProviderStatus.Conditions, configv1.AzurePlatformType)
+		}
+	default:
+		return false, errTestForPlatformNotImplemented
+	}
+	return false, nil
+}
+
+// hasInsufficientCapacity return true if there is an insufficient spot capacity condition.
+func hasInsufficientCapacityCondition(conditions []metav1.Condition, platform configv1.PlatformType) (bool, error) {
+	for _, condition := range conditions {
+		if condition.Type == string(machinev1.MachineCreation) && condition.Status == metav1.ConditionFalse {
```

Comment:
> condition.Type == string(machinev1.MachineCreation)

I believe `machinev1.MachineCreated` also should be counted here.
https://github.com/openshift/api/blob/master/machine/v1beta1/types_provider.go#L135 - see definition.

Reply:
You are right, broke this when refactoring.